### PR TITLE
Should not create PR for base branch in Auto-PR

### PR DESCRIPTION
**Pull request by brfrn169**
## Description

Currently, if we create a PR based on a release branch, Auto-PR creates a PR for the base branch in addition to other target branches. Since the target change is already merged to the base branch, it's unnecessary. This PR addresses this issue.

## Related issues and/or PRs

N/A

## Changes made

- Change not to create a PR for a base branch

## Checklist

- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have updated the documentation to reflect the changes.
- [x] Any remaining open issues linked to this PR are documented and up-to-date (Jira, GitHub, etc.).
- [x] Tests (unit, integration, etc.) have been added for the changes.
- [x] My changes generate no new warnings.
- [x] Any dependent changes in other PRs have been merged and published.

## Additional notes (optional)

I've tested it in my private repository.

## Release notes

N/A

### CODE REVIEW

File: ci/auto-pr/conv_proj_version_to_branch [hunk 1]
```diff
@@ -12,8 +12,7 @@ versions.inject(Set.new) {|acc, v|
       # e.g. project: "ScalarDB 4.0.0" -> branch: "master"
       #
       # This GitHub project corresponds to `main`/`master` branch.
-      # The target change is already merged to the default branch.
-      # Nonthing to do.
+      acc << "master"
```
Comment (komamitsu) [inline ci/auto-pr/conv_proj_version_to_branch:15]: Sorry, I didn't get why this change is needed. Could you tell me what would happen without the change?

Comment (brfrn169) [reply ci/auto-pr/conv_proj_version_to_branch:15]: This change actually doesn't affect the behavior. With the change in this PR, we remove the base branch of the original PR from the target branch list. So we no longer need the special care for the `main`/`master` branch in this `conv_proj_version_to_branch` script. What do you think?

Comment (komamitsu) [reply ci/auto-pr/conv_proj_version_to_branch:15]: I see. My concern is this default branch name is different among ScalarDB projects, and ideally it should be passed by `auto-pr.yaml` that gets the default branch name from `github.event.repository.default_branch`. But it takes some effort and introduces a bit of complexity. So it would be better to make it simple by removing this.

Comment (brfrn169) [reply ci/auto-pr/conv_proj_version_to_branch:15]: > My concern is this default branch name is different among ScalarDB projects

I thought the same thing, and for now, it seems we'll need to use slightly modified scripts (with different default branch names) in other repositories. In the future, we can unify the default branch name across the repositories. 

Moreover, considering that we create PRs from not just the default branches (`master` or `main`) but also from support/release branches (`3` and `3.x`), the following comment may not always apply:
```bash
      # The target change is already merged to the default branch.
      # Nonthing to do.
```

Given this, I feel the special handling to the `master`/`main` branches strange here. That's why I made the change.

What do you think?

Comment (komamitsu) [reply ci/auto-pr/conv_proj_version_to_branch:15]: Sounds reasonable to me 👍

**Review by komamitsu — APPROVED** [review]
LGTM! Thanks for handling the issue 👍
